### ISSUE TITLE
Assertion failure in `vim.lsp.buf.rename()` for pyright

### ISSUE THREAD

**Issue by anthony-S93** [neovim/nvim-lspconfig#2938]
### Description

Calling `vim.lsp.buf.rename()` when pyright is attached to the buffer causes the following assertion failure:

```
Error executing vim.schedule lua callback: /usr/share/nvim/runtime/lua/vim/uri.lua:107: URI m
ust contain a scheme: /home/anthony93/Devfiles/playground/testnvim/test.py
stack traceback:
        [C]: in function 'assert'
        /usr/share/nvim/runtime/lua/vim/uri.lua:107: in function 'uri_to_fname'
        /usr/share/nvim/runtime/lua/vim/uri.lua:128: in function 'uri_to_bufnr'
        /usr/share/nvim/runtime/lua/vim/lsp/util.lua:578: in function 'apply_text_document_ed
it'
        /usr/share/nvim/runtime/lua/vim/lsp/util.lua:866: in function 'apply_workspace_edit'
        /usr/share/nvim/runtime/lua/vim/lsp/handlers.lua:300: in function 'handler'
        /usr/share/nvim/runtime/lua/vim/lsp/buf.lua:321: in function 'handler'
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1393: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```
This error is not reproducible on other language servers I've tested (`jdtls`, `tsserver`, etc.). Only `pyright` is affected. I didn't have time to look into the details, but it may or may not have something to do with this microsoft/pyright#6519

### Neovim version

NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1702233742

### Operating system and version

Arch Linux 

### Affected language servers

pyright 1.1.340-1 (earlier versions of pyright aren't affected)

### Steps to reproduce

1. Write a simple function
```python
def foobar(parameter):
     print(parameter)
```
2. Try to rename any variable using the `vim.lsp.buf.rename()` function.

### Actual behavior

The operation resulted in an error. 

### Minimal config

```Lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({"config", "data", "state", "cache"}) do
    vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

--bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system({
        "git",
        "clone",
        "--filter=blob:none",
        "--single-branch",
        "https://github.com/folke/lazy.nvim.git",
        lazypath,
    })
end
vim.opt.runtimepath:prepend(lazypath)


--install plugins
local plugins = {
    {
        "nvim-treesitter/nvim-treesitter",
        build = ":TSUpdate",
    },

    {
        "neovim/nvim-lspconfig",
        config = function()
            local lspconfig = require("lspconfig")
            local util = require("lspconfig.util")
            local opts = { noremap = true, silent = true }

            --Keymaps on attach
            local on_attach = function(client, bufnr)
                vim.api.nvim_buf_set_option(bufnr, 'omnifunc', 'v:lua.vim.lsp.omnifunc')

                local bufopts = { noremap = true, silent = true, buffer = bufnr }
                vim.keymap.set('n', 'gD', vim.lsp.buf.declaration, bufopts)
                vim.keymap.set('n', 'gd', vim.lsp.buf.definition, bufopts)
                vim.keymap.set('n', 'K', vim.lsp.buf.hover, bufopts)
                vim.keymap.set('n', 'gi', vim.lsp.buf.implementation, bufopts)
                vim.keymap.set('n', '<C-k>', vim.lsp.buf.signature_help, bufopts)
                vim.keymap.set('n', '<space>wa', vim.lsp.buf.add_workspace_folder, bufopts)
                vim.keymap.set('n', '<space>wr', vim.lsp.buf.remove_workspace_folder, bufopts)
                vim.keymap.set('n', '<space>wl', function()
                    print(vim.inspect(vim.lsp.buf.list_workspace_folders()))
                end, bufopts)
                vim.keymap.set('n', '<space>D', vim.lsp.buf.type_definition, bufopts)
                vim.keymap.set('n', '<space>rn', vim.lsp.buf.rename, bufopts)
                vim.keymap.set('n', '<space>ca', vim.lsp.buf.code_action, bufopts)
                vim.keymap.set('n', 'gr', vim.lsp.buf.references, bufopts)
            end

            local lsp_flags = {
                debounce_text_changes = 150,
            }

            --Customize diagnostics Icons
            local signs = { Error = '󰅙', Warn = '󰀦', Hint = "󰐗", Info = "󰋼" }
            for type, icon in pairs(signs) do
                local hl = "DiagnosticSign" .. type
                vim.fn.sign_define(hl, { text = icon, texthl = hl, numhl = hl })
            end
            vim.diagnostic.config({
                underline = false,
                virtual_text = false,
            })

            lspconfig.pyright.setup({
                on_attach = on_attach,
            })
            
        end
    }
}

require("lazy").setup(plugins, { root = root .. "/plugins", })

--setup treesitter
local ok, treesitter = pcall(require, "nvim-treesitter.configs")
if ok then
    treesitter.setup({
        ensure_installed = { "python" },
        highlight = { enable = true },
    })
end
```

### LSP log

Not applicable (no logs were produced)

**Comment by gpanders** [comment]
I can't reproduce this, the rename function works just fine for me.

Please provide a _minimal_ reproduction that demonstrates the error (there should be no need to include `nvim-treesitter` or your custom mappings unless those are necessary to reproduce the error).

**Comment by volnushka** [comment]
I have the same issue.
Windows 10, WSL 2, no external plugins except lspconfig.

My `init.lua`:
```lua
-- Map leader key
vim.g.mapleader = ' '
vim.g.maplocalleader = ' '

-- Set search highlighting
vim.o.hlsearch = true

-- Set line numbers
vim.o.number = true

-- Set brake indent
vim.o.breakindent = true

-- Set undofile
vim.o.undofile = true

-- Set search case-sensitivity
vim.o.ignorecase = true
vim.o.smartcase = true

-- Set sign column
vim.wo.signcolumn = 'yes'

-- Set update time
vim.o.updatetime = 250
vim.o.timeoutlen = 300

-- For better completion experience
vim.o.completeopt = 'menuone,noselect'

-- Set terminal colors usage
vim.o.termguicolors = true

-- LSP
vim.lsp.set_log_level('debug')
local lspconfig = require('lspconfig')
lspconfig.pyright.setup({})

-- Global keymaps
vim.keymap.set({ 'n', 'v' }, '<Space>', '<Nop>', { expr = true, silent = true })
vim.keymap.set('n', 'k', "v:count == 0 ? 'gk' : 'k'", { expr = true, silent = true })
vim.keymap.set('n', 'j', "v:count == 0 ? 'gj' : 'j'", { expr = true, silent = true })
vim.keymap.set('n', '[d', vim.diagnostic.goto_prev, { desc = 'Go to previous diagnostic message' })
vim.keymap.set('n', ']d', vim.diagnostic.goto_next, { desc = 'Go to next diagnostic message' })
vim.keymap.set('n', '<leader>e', vim.diagnostic.open_float, { desc = 'Open floating diagnostic message' })
vim.keymap.set('n', '<leader>q', vim.diagnostic.setloclist, { desc = 'Open diagnostics list' })
	
vim.api.nvim_create_autocmd('LspAttach', {
	group = vim.api.nvim_create_augroup('UserLspConfig', {}),
	callback = function(ev)
		vim.bo[ev.buf].omnifunc = 'v:lua.vim.lsp.omnifunc'

		local opts = { buffer = ev.buf }
		-- Keymaps
		vim.keymap.set('n', 'gD', vim.lsp.buf.declaration, opts)
		vim.keymap.set('n', 'gd', vim.lsp.buf.definition, opts)
		vim.keymap.set('n', 'K', vim.lsp.buf.hover, opts)
		vim.keymap.set('n', 'gi', vim.lsp.buf.implementation, opts)
		vim.keymap.set('n', '<C-k>', vim.lsp.buf.signature_help, opts)
		vim.keymap.set('n', '<leader>D', vim.lsp.buf.type_definition, opts)
		vim.keymap.set('n', '<leader>rn', vim.lsp.buf.rename, opts)
		vim.keymap.set('n', 'gr', vim.lsp.buf.references, opts)
	end,
})
```

A test file:
```python
import os

a = os.path.exists("/")
```
When doing `<leader>rn` and renaming the variable `a` in the prompt:
```
Error executing vim.schedule lua callback: ...ocal/bin/nvim-linux64/share/nvim/runtime/lua/vim/uri.lua:107: URI must contain a scheme: /home/struchkov/test_nvim/test.py
stack traceback:
        [C]: in function 'assert'
        ...ocal/bin/nvim-linux64/share/nvim/runtime/lua/vim/uri.lua:107: in function 'uri_to_fname'
        ...ocal/bin/nvim-linux64/share/nvim/runtime/lua/vim/uri.lua:128: in function 'uri_to_bufnr'
        ...bin/nvim-linux64/share/nvim/runtime/lua/vim/lsp/util.lua:578: in function 'apply_text_document_edit'
        ...bin/nvim-linux64/share/nvim/runtime/lua/vim/lsp/util.lua:866: in function 'apply_workspace_edit'
        ...nvim-linux64/share/nvim/runtime/lua/vim/lsp/handlers.lua:300: in function 'handler'
        .../bin/nvim-linux64/share/nvim/runtime/lua/vim/lsp/buf.lua:321: in function 'handler'
        ...ocal/bin/nvim-linux64/share/nvim/runtime/lua/vim/lsp.lua:1394: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

**Comment by anthony-S93** [comment]
> I can't reproduce this, the rename function works just fine for me.
> 
> Please provide a _minimal_ reproduction that demonstrates the error (there should be no need to include `nvim-treesitter` or your custom mappings unless those are necessary to reproduce the error).

If you remove treesitter from the minimal config in my earlier post, the result will obviously be the same. The custom mapping is there to make it a bit easier to call the rename function. In any case, I did as you requested and further reduced the config to this: 

```lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({"config", "data", "state", "cache"}) do
    vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

--bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system({
        "git",
        "clone",
        "--filter=blob:none",
        "--single-branch",
        "https://github.com/folke/lazy.nvim.git",
        lazypath,
    })
end
vim.opt.runtimepath:prepend(lazypath)


--install plugins
local plugins = {
    {
        "neovim/nvim-lspconfig",
        config = function()
            local lspconfig = require("lspconfig")
            local util = require("lspconfig.util")
            local opts = { noremap = true, silent = true }

            --Keymaps on attach
            local on_attach = function(client, bufnr)
                vim.api.nvim_buf_set_option(bufnr, 'omnifunc', 'v:lua.vim.lsp.omnifunc')
                local bufopts = { noremap = true, silent = true, buffer = bufnr }
                vim.keymap.set('n', '<space>rn', vim.lsp.buf.rename, bufopts)
            end

            lspconfig.pyright.setup({
                on_attach = on_attach,
            })
            
        end
    }
}

require("lazy").setup(plugins, { root = root .. "/plugins", })
```
The error still occurs. Unsurprisingly. I must once again reiterate that the error only occurs for `pyright`. And for that reason, I suspect it has something to do with microsoft/pyright#6519

**Comment by anthony-S93** [comment]
Turns out this is an issue with `pyright`, which is fixed in version `1.1.342`
